lib/src/hitmap.dart

```diff
@@ -80,7 +95,9 @@ Future<Map<String, Map<int, int>>> createHitmap(
     }
 
     final sourceHitMap = globalHitMap.putIfAbsent(source, () => <int, int>{});
-    final hits = e['hits'] as List;
+    var hits = e['hits'] as List;
+    // Ignore line annotations require hits to be sorted.
+    hits = _sortHits(hits);
```

Comment:
Since you're going to the trouble of parsing `hits` into `structuredHits`, which is a much nicer format, I think it would make more sense to leave it in that format for the remainder of this function, rather than transforming it back into the weird flat array.

So in the `_HitInfo` class you could remove `hitRange` and add a `int lastLine` (and set `lastLine` to be the same as `firstLine` in the case that `hitRange` is an `int`).

Reply:
Would it be okay to do that in a separate PR?
I feel that making that change here would be a bit too much.
Changing the underlying data structure was definitely not the main objective of the change.

Reply:
I think you'll only have to delete a few lines of code from the loop below, and tweak _sortHits a bit. But I suppose it's fine to do that in a follow up.